app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -198,8 +200,11 @@ class SearchFragment : Fragment(), UserInteractionHandler {
             qrFeature.get()?.scan(R.id.container)
         }
 
-        view.back_button.setOnClickListener {
-            findNavController().navigateUp()
+        view.back_button.apply {
+            setOnClickListener {
+                animateBackButtonAway()
+                findNavController().navigateUp()
```

Comment:
I think it's a bad sign that this animates then navigates up, while the ToolbarView calls the interactor, which calls the controller, which... animates then navigates up.  This is one of the things our architecture is meant to unify, so let's clean it up while we're here.

If we:
1) move `animateBackButtonAway` off of `ToolbarView` and on to `SearchController`
2) replace 205 & 206 in `SearchFragment` with `searchInteractor.onEditingCanceled()`

then we have a more reasonable and consistent data flow.

It seems to me that we want back press and back button clicks to behave the same here, so I'm happy to have them call the same interactor method.  If that ever changes in the future, we can break them out as needed.